### PR TITLE
feat: Added methods for creating and deleting composite indexes

### PR DESCRIPTION
**Pull request by parthea**
This PR fixes an issue with the `owlbot.py` file which prevented [google/cloud/datastore_v1](https://github.com/googleapis/python-datastore/commits/main/google/cloud/datastore_v1) from being updated. This PR also brings the client up to date with the generated client in googleapis-gen [here](https://github.com/googleapis/googleapis-gen/tree/master/google/datastore).

chore: update owlbot.py to properly copy folders from googleapis-gen
fix(deps): require google-api-core >= 1.28.0
docs: list oneofs in docstring
feat: add context manager support in client
fix: add 'dict' annotation type to 'request'
fix: add async client to %name_%version/init.py
feat: support self-signed JWT flow for service accounts 
feat: Added methods for creating and deleting composite indexes

### CODE REVIEW

**Comment by parthea** [comment]
This is ready for review.

File: owlbot.py [hunk 1]
```diff
@@ -13,41 +13,80 @@
...
+
 import synthtool as s
 from synthtool import gcp
 from synthtool.languages import python
 
 common = gcp.CommonTemplates()
 
+# This is a customized version of the s.get_staging_dirs() function from synthtool to
+# cater for copying 2 different folders from googleapis-gen
+# which are datastore and datastore/admin
```
Comment (busunkim96) [inline owlbot.py:27]: [spanner](https://github.com/googleapis/python-spanner/tree/main/google/cloud) and [firestore](https://github.com/googleapis/python-firestore/tree/main/google/cloud), [bigtable](https://github.com/googleapis/python-bigtable/tree/main/google/cloud) all have a similar layout with a main client and at least one other client. Is it worth generalizing this function in `synthtool` instead?

File: google/cloud/datastore_admin_v1/services/datastore_admin/client.py [hunk 2]
```diff
@@ -646,40 +658,175 @@ def import_entities(
...
 
+    def create_index(
+        self,
+        request: Union[datastore_admin.CreateIndexRequest, dict] = None,
+        *,
+        retry: OptionalRetry = gapic_v1.method.DEFAULT,
+        timeout: float = None,
+        metadata: Sequence[Tuple[str, str]] = (),
+    ) -> operation.Operation:
+        r"""Creates the specified index. A newly created index's initial
```
Comment (busunkim96) [inline google/cloud/datastore_admin_v1/services/datastore_admin/client.py:676]: This looks like a brand new API method - should it be mentioned in the release notes?

File: google/cloud/datastore_admin_v1/services/datastore_admin/client.py [hunk 3]
```diff
@@ -646,40 +658,175 @@ def import_entities(
...
+        return response
+
+    def delete_index(
+        self,
+        request: Union[datastore_admin.DeleteIndexRequest, dict] = None,
+        *,
+        retry: OptionalRetry = gapic_v1.method.DEFAULT,
+        timeout: float = None,
+        metadata: Sequence[Tuple[str, str]] = (),
+    ) -> operation.Operation:
```
Comment (busunkim96) [inline google/cloud/datastore_admin_v1/services/datastore_admin/client.py:744]: Same here, this looks like a new API method

**Review by busunkim96 — APPROVED** [review]

File: owlbot.py (hunk 1 above)
Comment (parthea) [reply owlbot.py:27]: I opened a PR in the synthtool repo to propose this change but I was not able to get it approved. https://github.com/googleapis/synthtool/pull/1107

File: google/cloud/datastore_admin_v1/services/datastore_admin/client.py (hunk 2 above)
Comment (parthea) [reply google/cloud/datastore_admin_v1/services/datastore_admin/client.py:676]: Great catch! Done.

**Pull request merged**